crates/dojo-core/src/world/world_contract.cairo

```diff
@@ -23,8 +32,12 @@ pub trait IWorld<T> {
 
     fn register_namespace(ref self: T, namespace: ByteArray);
 
-    fn register_model(ref self: T, class_hash: ClassHash);
-    fn upgrade_model(ref self: T, class_hash: ClassHash);
+    fn register_model(ref self: T, model_definition: dojo::model::ModelDefinition);
+    fn upgrade_model(ref self: T, model_selector: felt252, version: u8, ty: Ty, layout: Layout);
+
+    fn model_ty(self: @T, model_selector: felt252) -> Ty;
+    fn model_layout(self: @T, model_selector: felt252) -> Layout;
+    fn model_info(self: @T, model_selector: felt252) -> ModelInfo;
```

Comment:
We may not need those 3 if we have the model's definition returned by `resource`? Or do you have something else in mind?

Reply:
You're right. When modifiying sozo commands and the migration process I will see which functions are really useful. The idea is to have some "shortcut" functions to avoid reading the whole model definition if we just need model info like namespace or name.